### PR TITLE
Refactor/car controller

### PR DESCRIPTION
**Pull request by DJSteins**
Refactors CarController into CarControllerBehaviour and CarController. Adds unit tests for CarController.

### CODE REVIEW

**Comment by DJSteins** [comment]
[Refactor CarController for testing](https://trello.com/c/RQbrvXrQ/314-refactor-carcontroller-for-testing)

File: Assets/_PartyParkade/Script/Game/Car/CarControllerBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+        /// <summary>
+        /// Applies a tilt to the rigidbody when the car is turning.
+        /// </summary>
+        private void WheelBodyTilt()
+        {
+            // Wheel and body tilt
+            Quaternion quart = m_carController.GetWheelTilt();
+            m_carController.SetWheelTilt(m_wheelFrontLeft, quart);
+            m_carController.SetWheelTilt(m_wheelFrontRight, quart);
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/Car/CarControllerBehaviour.cs:177]: Can you combine the two methods into 1?,
SetWheelTile(left, right, quart)

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/Car/CarControllerBehaviour.cs:177]: wait you get the quart from m_carController.GetWheelTilt();

can you just call one method, SetWheelTilt(left, right) and the quart is just inside the carcontroller

Comment (DJSteins) [reply Assets/_PartyParkade/Script/Game/Car/CarControllerBehaviour.cs]: I want to do left then right so I don't have to duplicate the "is wheel null" code

**Pull request merged**